testing/test_metal_context.mm

```diff
@@ -42,6 +42,9 @@
 
 TestMetalContext::~TestMetalContext() {
   std::scoped_lock lock(textures_mutex);
+  for (auto& [_, texture] : textures_) {
+    [(id)texture.get() release];
+  }
```

Comment:
This isn't necessary, those textures are stored in sk_cfp which calls release in its deconstructor.  If you see this being necessary you probably have a double retain.  It looks correct when you are making it to me, the apis are a bit confusing though.

Reply:
Ah right - I missed that we were calling the `sk_cfp` copy ctor when assigning the `MTLTexture` into the `textures_` map (and that the `textures_` map holds the `sk_cfp` and not a raw pointer). This means that we don't need to release manually in this dtor, but I believe we still need to remove the extra `retain` call, right?

Reply:
Nope, I believe we should have the retain now.  `[device newTextureWithDescriptor:texture_descriptor.get()]` should return an object that is on the autorelease pool.  Which means it will get released when this event is done, so we'd need a retain to keep it alive in the smart pointer.

I'm not 100% certain because the semantics for an objc method called `new*` that return an objc object are a bit unclear to me.  I think technically anything that isn't named `create` should be on the autorelease pool, but that isn't the case for `-[NSObject new]`.

Reply:
Unless I'm misunderstanding the guide at https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/MemoryMgmt/Articles/mmRules.html, it says that any ObjC method with "new" in the name returns an object that the caller owns outright. That implies to me that it won't be in the autorelease pool.

That being said, I think that the `MTLTextureDescriptor` *does* fall under that category perhaps, as the ObjC method name does not include the word "new", "create" etc, so I'd expect that to be in the autorelease pool?

Reply:
Ah, yep, you're right, anything that begins with "new" means the caller owns it.  No need to retain it again.  You don't run into `new` methods often.

Reply:
OK, the unit test that was failing was because the `MTLTextureDescriptor` is indeed returned already in the autorelease pool, so that retain still needs to be there.

With that fixed, all the unit tests are now passing locally.